pkg/resource/deploy/plan.go

```diff
@@ -117,6 +117,7 @@ type Plan struct {
 	preview              bool                             // true if this plan is to be previewed rather than applied.
 	depGraph             *graph.DependencyGraph           // the dependency graph of the old snapshot
 	providers            *providers.Registry              // the provider registry for this plan.
+	current              map[resource.URN]*resource.State // a map of all current resources
```

Comment:
surprised this isn't avialable off of the Plan (or some other engine object already).  @pgavlin ?

Reply:
Yeah - I talked to @pgavlin about this when I was initially trying to get this to work.  There did not appear to be any simpler way to access this - and the alternative he suggested was a more complex option of registering to get the resource state serialization events and manage storage of them ourselves in the built-in.  The approach I'm taking here has appeared to work fine so far - but if @pgavlin tells me we should do something else I'm happy to change.

Reply:
I don't like over-exposing this, but ultimately I think it'll be fine. We should make this a `sync.Map` s.t. it is safe to access in parallel. We should also note that a resource only appears in this map _after_ its registration has completed: a `read` that is received before a registration completes will "happen before" the registration, as it won't be able to observe the resource.